src/proxy/routes-engine.ts

```diff
@@ -3,34 +3,62 @@ import http from "http";
 import { DEFAULT_CONFIG } from "../config";
 import { decodeCookie, findFile } from "../core/utils";
 
-export const handleUserCustomRoutes = async (folder: string): Promise<UserDefinedRoute[]> => {
+export const handleUserConfig = async (folder: string): Promise<UserConfig> => {
   if (!fs.existsSync(folder)) {
-    return [];
+    return { userDefinedRoutes: [] };
```

Comment:
Return userConfig with an empty userDefinedRoutes array as that is the only required field and it's the same behavior as before.